### PR TITLE
Merge 1-configure-frontend-repository into master

### PR DESCRIPTION
**Pull request by macszl**
Configuration of our frontend, including a default Vite TS+React configuration and the required ESLint and Prettier configs.

### CODE REVIEW

**Review by kaczor6418 — CHANGES_REQUESTED** [review]
I think we do not need swc as we choose Vite over Webpack. In general looks good but require few fixes. In upcoming days I am going to post eslint and typescript config files

File: README.md [hunk 1]
```diff
@@ -1 +1,36 @@
...
+  ```bash
+  git clone https://github.com/dorm-social-room-manager/dsrm-frontend.git
+  cd ./dsrm-frontend
+  ``
+## Run Locally
+
+To run the application locally, start the server using the following command
+
+```bash
+  npm run dev
```
[the remark this answers is not in the document]
Comment (macszl) [reply README.md]: Fixed in d6dfc02

File: package.json [hunk 2]
```diff
@@ -0,0 +1,44 @@
+{
+  "name": "inno-test",
+  "private": true,
+  "version": "0.0.0",
```
[the remark this answers is not in the document]
Comment (macszl) [reply package.json]: Fixed in 969702c

File: package.json [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  "type": "module",
+  "scripts": {
+    "build": "swc src -d dist && vite build",
```
[the remark this answers is not in the document]
Comment (macszl) [reply package.json]: Skipped using swc, so this should be fixed in 969702c

File: package.json [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    "preview": "vite preview",
+    "start": "vite",
+    "lint": "eslint . --ext .ts",
```
[the remark this answers is not in the document]
Comment (macszl) [reply package.json]: Added support for .tsx in 969702c

File: package.json [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    "lint:fix": "eslint . --ext .ts --fix",
```
[the remark this answers is not in the document]
Comment (macszl) [reply package.json]: Added support for .tsx in 969702c

File: package.json [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+  },
+  "dependencies": {
+    "react": "^18.2.0",
+    "react-dom": "^18.2.0"
+  },
+  "devDependencies": {
+    "@emotion/react": "^11.10.5",
+    "@emotion/styled": "^11.10.5",
+    "@eslint/create-config": "^0.4.1",
+    "@mui/material": "^5.10.14",
```
[the remark this answers is not in the document]
Comment (macszl) [reply package.json]: Moved MaterialUI component library to prod dependencies in 969702c

File: vite.config.ts [hunk 7]
```diff
@@ -0,0 +1,7 @@
+import { defineConfig } from 'vite'
+import react from '@vitejs/plugin-react'
+
+// https://vitejs.dev/config/
```
[the remark this answers is not in the document]
Comment (macszl) [reply vite.config.ts]: Removed the comment in 969702c

File: package.json [hunk 8]
```diff
@@ -0,0 +1,44 @@
+{
+  "name": "inno-test",
```
Comment (kaczor6418) [inline package.json:2]: should be dsrm-forntend

File: package.json [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+  "private": true,
+  "version": "1.0.0",
+  "type": "module",
+  "scripts": {
```
Comment (kaczor6418) [inline package.json:6]: lint:watch is missing

File: .eslintrc.json [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+    "browser": true,
+    "node": true
+  },
+  "plugins": ["prettier"],
+  "extends": ["eslint:recommended", "plugin:prettier/recommended"],
+  "rules": {
+    "no-eval": 2,
+    "prefer-const": 2
+  }
+}
```
Comment (kaczor6418) [inline .eslintrc.json:17]: please update with *eslintconfig* from issue description

File: tsconfig.json [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+    "module": "ESNext",
+    "moduleResolution": "Node",
+    "resolveJsonModule": true,
+    "isolatedModules": true,
+    "noEmit": true,
+    "jsx": "react-jsx"
+  },
+  "include": ["src"],
+  "references": [{ "path": "./tsconfig.node.json" }]
+}
```
Comment (kaczor6418) [inline tsconfig.json:21]: Please update with tsconfig from issue description

**Review by kaczor6418 — CHANGES_REQUESTED** [review]
Still needs improvements

File: package.json (hunk 8 above)
Comment (235923) [reply package.json]: Fixed in d3482ab3cd3bc0279ad198369012a9fd8b289138

File: .eslintrc.json (hunk 10 above)
Comment (235923) [reply .eslintrc.json]: Updated in d3482ab3cd3bc0279ad198369012a9fd8b289138

File: tsconfig.json (hunk 11 above)
Comment (235923) [reply tsconfig.json]: Updated in d3482ab3cd3bc0279ad198369012a9fd8b289138

File: package.json (hunk 9 above)
Comment (macszl) [reply package.json:6]: Required functionality added in f97ce5e

File: tsconfig.node.json [hunk 12]
```diff
@@ -0,0 +1,9 @@
+{
+  "compilerOptions": {
+    "composite": true,
+    "module": "ESNext",
+    "moduleResolution": "Node",
+    "allowSyntheticDefaultImports": true
+  },
```
[the remark this answers is not in the document]
Comment (macszl) [reply tsconfig.node.json]: Should be okay now, thanks to 40db606

File: .eslintignore [hunk 13]
```diff
@@ -0,0 +1,5 @@
+/node_modules/*
+/dist/*
+.eslintrc.json
```
Comment (kaczor6418) [inline .eslintignore:3]: You do not need to include `.json` files because you run eslint vs `.ts` and `.tsx` files

File: .prettierrc [hunk 14]
```diff
@@ -0,0 +1,18 @@
+{
+  "printWidth": 150,
+  "tabWidth": 2,
```
Comment (kaczor6418) [inline .prettierrc:3]: Can we call this `.prettierrc.json` to be consistent between all configuration files?

File: package.json [hunk 15]
```diff
@@ -0,0 +1,47 @@
+{
+  "name": "dsrm-frontend",
+  "private": true,
```
Comment (kaczor6418) [inline package.json:3]: Why private? Please remove

File: package.json [hunk 16]
```diff
@@ -0,0 +1,47 @@
...
+    "@emotion/react": "^11.10.5",
+    "@emotion/styled": "^11.10.5",
+    "@mui/material": "^5.10.15",
+    "react": "^18.2.0",
+    "react-dom": "^18.2.0",
+    "sass": "^1.56.1"
+  },
+  "devDependencies": {
+    "@eslint/create-config": "^0.4.1",
+    "@swc/cli": "^0.1.57",
```
Comment (kaczor6418) [inline package.json:26]: Let's remove swc dependencies as we are not using it now

**Review by kaczor6418 — CHANGES_REQUESTED** [review]

File: package.json (hunk 16 above)
Comment (macszl) [reply package.json]: Removed in 2fe809c

File: .prettierrc (hunk 14 above)
Comment (macszl) [reply .prettierrc]: Changed in 2fe809c

File: .eslintignore (hunk 13 above)
Comment (macszl) [reply .eslintignore]: Removed the .eslintignore .json files in 2fe809c

File: package.json (hunk 15 above)
Comment (macszl) [reply package.json]: Changed to "private":false in 2fe809c

**Review by kaczor6418 — CHANGES_REQUESTED** [review]
LGTM :sunglasses:

**Review by kaczor6418 — APPROVED** [review]
LGTM :sunglasses:

**Pull request merged**